data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/actions/MergeAllAggregateAction.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.aggregate.actions;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateAction;
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateActionInput;
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateActionResponse;
+import org.opensearch.dataprepper.plugins.processor.aggregate.GroupState;
+
+/**
+ * An AggregateAction that combines multiple Events into a single Event. This
+ * action will add the unique keys of each smaller Event to the overall
+ * groupState,
+ * and will create a combined Event from the groupState on concludeGroup. If
+ * smaller Events have the same keys, then these keys will be overwritten with
+ * the keys of the
+ * most recently handled Event.
+ * 
+ * @since 1.3
```

Comment:
This should be `@since 2.1` (if we get in this week). Otherwise it will be `@since 2.2`.

Reply:
Made it 2.2